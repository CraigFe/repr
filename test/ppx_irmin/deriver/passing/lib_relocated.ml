module Elsewhere : sig
  module Foo : module type of Irmin_type.Type

  type t [@@deriving irmin { lib = Some "Foo" }]
end = struct
  module Foo = Irmin_type.Type

  module Irmin = struct end

  type t = unit * unit [@@deriving irmin { lib = Some "Foo" }]
end

module Locally_avaliable : sig
  type 'a ty

  type t [@@deriving irmin { lib = None }]
end = struct
  let pair, unit = Irmin_type.Type.(pair, unit)

  type 'a ty = 'a Irmin_type.Type.ty

  module Irmin = struct end

  type t = unit * unit [@@deriving irmin { lib = None }]
end
