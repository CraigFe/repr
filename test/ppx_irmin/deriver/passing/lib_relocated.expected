module Elsewhere :
  sig
    module Foo : module type of Irmin_type.Type
    type t[@@deriving irmin { lib = (Some "Foo") }]
    include sig val t : t Foo.t end[@@ocaml.doc "@inline"][@@merlin.hide ]
  end =
  struct
    module Foo = Irmin_type.Type
    module Irmin = struct  end
    type t = (unit * unit)[@@deriving irmin { lib = (Some "Foo") }]
    include struct let t = Foo.pair Foo.unit Foo.unit end[@@ocaml.doc
                                                           "@inline"]
    [@@merlin.hide ]
  end 
module Locally_avaliable :
  sig
    type 'a ty
    type t[@@deriving irmin { lib = None }]
    include sig val t : t ty end[@@ocaml.doc "@inline"][@@merlin.hide ]
  end =
  struct
    let (pair, unit) = let open Irmin_type.Type in (pair, unit)
    type 'a ty = 'a Irmin_type.Type.ty
    module Irmin = struct  end
    type t = (unit * unit)[@@deriving irmin { lib = None }]
    include struct let t = pair unit unit end[@@ocaml.doc "@inline"][@@merlin.hide
                                                                    ]
  end 
